editor/gui/editor_file_dialog.cpp

```diff
@@ -64,8 +64,9 @@ void EditorFileDialog::popup_file_dialog() {
 void EditorFileDialog::_focus_file_text() {
 	int lp = file->get_text().rfind(".");
 	if (lp != -1) {
+		file->grab_focus_edit();
 		file->select(0, lp);
-		file->grab_focus();
+		file->set_caret_column(lp);
```

Comment:
need to keep the caret at the start of the `LineEdit` too, because pressing left button to move the caret will move it to the start of the selection and will remove the selection.